### DISCUSSION TITLE
Problems with devices after replacement CC2531 with CC2652RB

### DISCUSSION THREAD

**Discussion by peez80** [Koenkk/zigbee2mqtt#8759] [General]
Because I had the feeling that the old CC2531 adapter is reaching it's bounds, I ordered a new CC2652RB adapter from the recommendation on the zigbee2mqtt homepage.

At first I tried to replace it transparently, by keeping ieee address, network key, pan ids etc.
This kinda worked half - half. The device availability felt flappy. The devices were randomly accessible or not.

Then I decided to completely rebuild the network by generating a new pan_id as well as a network key. Only the ieee address of the adapter stayed the same because I didn't want to flash again.

Well now - after pairing all devices, it still feels strange, yet not so much as before. The powered devices are rarely not reachable (I see ping errors in zigbee2mqtt log).
But battery powered devices are having hube trouble. I have some Ikea remotes, some Philips Hue Remotes, some Xiaomi temperature sensors and a pir sensor. All of these seem to pair correctly, work for some time but then just don't work anymore.
Even more, when pairing one of the Ikea remotes, zigbee2mqtt log keeps telling me that the device left the network over and over.

Any idea if the adapter might be damaged or did I miss something during adapter replacement?

**Comment by huub66** [comment]
I have done exactly the same migration and facing stability issues as well. It even seems that the CC2652RB is less stable compared to the CC2531 I had previously installed. Unfortunately haven't' found a solution yet, have re-flashed the adapter as well in addition to your actions, to no avail up to now..

**Comment by lux73** [comment]
can't image that...   after my Switch from cc2531 to cc2652rb all my Problems are gone...

are you both using an USB Extension Cable? **Please don't stick it directly in the USB Port** - it's recommended to using Extension Cable

**Reply by huub66** [reply]
Yes I am using a 1m extension cable. Funny thing is that I have never used an extension cable with the cc2531 and never had interference issues when using on my NUC.

**Comment by peez80** [comment]
@huub66 , honestly I am not sure what exactly solved my problems. This is what I remember I did (meanwhile the network is definitely more stable than with the old CC2531).

* What I saw as I described that some devices - especially the battery powered devices - joined and left the network over and over
* As a result the batteries were drained, which looked like another strange behaviour at first, because I recognized the empty batteries way later.
* Also what I did before is just pressing the "announce" button (once) on the xiaomi devices. This seemed to pair the device according to the logs, however the problem remained.
* At the end I changed the batteries and did a full reset at all devices that didn't work (e.g. pressing the button at the xiaomi devices for 4 secs).
* Additionally I had the Adapter (without extension cable) inside the Pi behind my TV, directly near the WiFi router and plenty of cables or wifi receivers. I moved the pi to another place (still without extension cable, just because I don't have one).

Now my network is better than ever. I have no clue what finally made the trick, but maybe there is something that might help you as well.

**Reply by huub66** [reply]
OK, think I'll just start from scratch with updated z2m and repairing all devices to see where that brings me
